libraries/standard/mqtt/src/iot_mqtt_operation.c

```diff
@@ -838,7 +844,29 @@ void _IotMqtt_ProcessKeepAlive( IotTaskPool_t pTaskPool,
     if( pPingreqOperation->u.operation.periodic.ping.nextPeriodMs ==
         pPingreqOperation->u.operation.periodic.ping.keepAliveMs )
     {
-        status = _sendPingRequest( pMqttConnection );
+        /* Only send the PINGREQ if the keep-alive period has elapsed since the connection
```

Comment:
Looks like we are skipping the keepalive here if elapsed time does not exceed the set value, but what if we JUST miss it, e.g. what if we have keepalive on 60s and we send every 60s.

Does this timer fire every 60s then? Because if it does this code will delay the NEXT PINGREQ by an additional 60s, which means if we were at 59s when this timer fired that means the next one will only be sent at 119s?

If this is not the case and the timing works differently then this is not a problem at all of course.

Reply:
Your understanding is correct.

Reply:
So then just make sure that we do not exceed the keepalive time because of this. I would expect us to instead of scheduling the timer after an additional PERIOD ms to instead schedule the timer to expire after REMAINING time in the current period, so PERIOD-elapsedTime ...

Reply:
Changed.